apps/www/src/forms/plugins/autoName.js

```diff
@@ -0,0 +1,105 @@
+import sh1 from '@hydrofoil/shaperone-core/ns.js'
+import { SELECT } from '@tpluscode/sparql-builder'
+import { rdfs, rdf, schema } from '@tpluscode/rdf-ns-builders'
+import { isGraphPointer, isLiteral } from 'is-graph-pointer'
+import { findNodes, toSparql } from 'clownface-shacl-path'
+import { view, viewBuilder } from '@view-builder/core/ns.js'
+import { store } from '../../state/store.js'
+
+export const autoResourceLabel = {
+  model: {
+    effects({ getDispatch }) {
+      const { forms } = getDispatch()
+
+      return {
+        /**
+         * When setting properties annotated with sh1:setFocusNodeLabel,
+         * set focus node's rdfs:label from the selected object
+         */
+        'forms/setObjectValue': async (arg) => {
+          const { pointer } = arg.property
+          const setFocusNodeLabel = pointer.out(sh1.setFocusNodeLabel)
+          const labelPath = pointer.out(sh1.labelPath)
+
+          if (!isGraphPointer(setFocusNodeLabel)) {
+            return
+          }
+
+          let { focusNode } = arg
+          if (!isLiteral(setFocusNodeLabel)) {
+            focusNode = findNodes(focusNode, setFocusNodeLabel)
+          }
+
+          const label = await loadResourceLabel(arg.newValue, labelPath)
+          focusNode.deleteOut(rdfs.label).addOut(rdfs.label, label)
+          forms.updateComponentState({ ...arg, newState: { label } })
+        },
+      }
+    },
+  },
+}
+
+export const filterLabel = {
+  model: {
+    effects({ getDispatch }) {
+      const { forms } = getDispatch()
+
+      return {
+        /**
+         * When setting properties of view:Filter,
+         * set the label based on its representation
+         */
+        'forms/setObjectValue': async (arg) => {
+          const { focusNode, property } = arg
+          if (focusNode.has(rdf.type, view.Filter).terms.length === 0) {
+            return
+          }
+
+          let label = 'Filter'
+          const dimension = focusNode
+            .out(viewBuilder.baseDimension)
+            .out(view.from)
+            .out(view.path)
+          if (isGraphPointer(dimension)) {
+            const filterTermSet = focusNode.out(viewBuilder.filterTermSet)
+            const operator = focusNode.out(view.operation)
+            const argument = focusNode.out(view.argument)
+
+            if (isGraphPointer(filterTermSet)) {
+              const dimensionLabel = await loadResourceLabel(dimension.term, schema.name)
+              const termSetLabel = await loadResourceLabel(filterTermSet.term, schema.name)
+
+              label = `${dimensionLabel} is one of "${termSetLabel}"`
+            } else if (isGraphPointer(operator) && isGraphPointer(argument)) {
+              const dimensionLabel = await loadResourceLabel(dimension.term, schema.name)
+              const operatorLabel = property.pointer.node(operator).out(rdfs.label).value
```

Comment:
Just curious, what is the criterion to choose between `schema:name` and others is `rdfs:label`.  ?

Reply:
The data in our stores use `schema:name`. Hence that is used in the query

`rdfs:label` is the default to display labels in a dropdown. (for now operators are [part of the shape graph](https://github.com/zazuko/ssz-view-builder/blob/main/apps/www/src/shapes/view.turtle#L36-L41))